src/pages/Authentication/Login.jsx

```diff
@@ -1,39 +1,109 @@
 import "../../styles/login.css";
-import { Link } from "react-router-dom";
-import{FaEyeSlash}from "react-icons/fa";
-export const Login =()=>{
-    return (<div className="login-container wd-100-pc pos-rel flex-grow display-flex-column">
-    <h1 className="auth-form-title">Login</h1>
-    <form action="auth-form" className="wd-100-pc display-flex-column">
+import React, { useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import {
+  initialLogInData,
+  loginAsGuestCredentials,
+} from "../../constants/index";
+import { useAuthentication } from "../../context/index";
+import { LoginActionHandler } from "../../services/index";
+
+export const Login = () => {
+  const navigate = useNavigate();
+  const { authDispatch } = useAuthentication();
+  const [loginData, setLoginData] = useState(initialLogInData);
+  const LogInDataHandler = (e) => {
+    const { name, value } = e.target;
+    setLoginData((prevLoginData) => ({ ...prevLoginData, [name]: value }));
+  };
+  return (
+    <div className="login-container wd-100-pc pos-rel flex-grow display-flex-column">
+      <h1 className="auth-form-title">Login</h1>
+      <form
+        action="auth-form"
+        className="wd-100-pc display-flex-column"
+        onSubmit={(e) => {
+          LoginActionHandler(e, loginData, authDispatch, navigate);
```

Comment:
No need to pass `e` to the function, you can do like this
```
onSubmit={(e) => {
          e.preventDefault()
          LoginActionHandler( loginData, authDispatch, navigate);

```

Reply:
Thanks for all sugesstions.